blog/2024-09-17-wasmcloud-postgres-capability-brings-pluggable-databases/index.mdx

```diff
@@ -0,0 +1,164 @@
+
+## Get started with Postgres on wasmCloud
+
+Let's try it out. We'll use a Rust-based component, but you don't really have to be familiar with the language. For this example, you'll just need:
+
+- The [wasmcloud Shell (`wash`)](https://wasmcloud.com/docs/installation) CLI to build WebAssembly components and deploy to local wasmCloud.
+- [`cargo`](https://doc.rust-lang.org/cargo/getting-started/installation.html) (part of the Rust toolchain) to help build from Rust
+- [`docker`](https://docs.docker.com) to easily run an instance of Postgres.
+
+### Start Postgres and wasmCloud locally
+
+First we need to get a Postgres cluster running. We can do this quickly and easily with Docker:
+
+```console
+docker run \
+    --rm \
+    -e POSTGRES_PASSWORD=postgres \
+    --name pg -p 5432:5432\
+    postgres:16.2-alpine
+```
+
+Next we'll start our local wasmCloud, using the `-d` flag to run in the background:
+
+```shell
+wash up -d
+```
+
+### Generate a new project
+
+Now we'll generate a new project using the Rust-based `sqldb-postgres-query` example from the wasmCloud monorepo as a template.
+
+```shell
+wash new component postgres-demo --git wasmcloud/wasmcloud --subfolder examples/rust/components/sqldb-postgres-query
+```
+```shell
+cd postgres-demo
+```
+
+If you're familiar with Rust, in `src/lib.rs` you'll see that the [code](https://github.com/wasmCloud/wasmCloud/blob/main/examples/rust/components/sqldb-postgres-query/src/lib.rs) for  querying Postgres is very simple:
+
+```rs
+impl Guest for QueryRunner {
+    fn call() -> String {
+        if let Err(e) = query(CREATE_TABLE_QUERY, &[]) {
+            return format!("ERROR - failed to create table: {e}");
+        };
+
+        match query(
+            INSERT_QUERY,
+            &[PgValue::Text(format!("inserted example row!"))],
+        ) {
+            Ok(rows) => format!("SUCCESS: inserted new row:\n{rows:#?}"),
+            Err(e) => format!("ERROR: failed to insert row: {e}"),
+        }
+    }
+}
+```
+
+Next build your WebAssembly component from the Rust code.
+
+```console
+wash build
+```
+
+The build process generates a folder called `build` containing `sqldb-postgres-query_s.wasm`.
+
+
+### Deploy the demo
+
+The next step for a wasmCloud application deployment is usually to define the components and providers you'll be using in a deployment manifest. We have a pre-populated manifest (`wadm.yaml`) in our project directory.
+
+In the interest of security, we'll use the `wash config` subcommand to define a "named configuration" called `default-postgres`: 
+
+```console
+wash config put default-postgres \
+    POSTGRES_HOST=localhost \
+    POSTGRES_PORT=5432 \
+    POSTGRES_USERNAME=postgres \
+    POSTGRES_PASSWORD=postgres \
+    POSTGRES_DATABASE=postgres \
+    POSTGRES_TLS_REQUIRED=false
+```
+
+In our application manifest (`wadm.yaml`), we tell our provider to simply use the named configuration that we provided.
```

Comment:
```suggestion
The Postgres provider knows to use this configuration because of our application maifest (`local.wadm.yaml`), where we've referred to `default-postgres` for named configuration.
```